cmd/cli/version/version.go

```diff
@@ -104,13 +107,71 @@ func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
 	if !oV.ClientOnly {
 		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
 		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
+			return fmt.Errorf("error running version: %w", err)
+		} else {
+			versions.ServerVersion = serverVersion
+			columns = append(columns, serverVersionColumn)
 		}
+	}
+
+	// Add additional information
+	versions.OperatingSystem = runtime.GOOS
+	versions.Architecture = runtime.GOARCH
+	versions.UserID = "some-user-id" // Replace this with your method to get UserID
 
-		versions.ServerVersion = serverVersion
-		columns = append(columns, serverVersionColumn)
+	var serverVersion string
+	if versions.ServerVersion != nil {
+		serverVersion = versions.ServerVersion.GitVersion
 	}
 
+	checkForUpdates(ctx, versions.ClientVersion.GitVersion, serverVersion, versions.OperatingSystem, versions.Architecture, versions.UserID)
```

Comment:
Alternatively, this method could just accept a `ctx` and `version` - pulling the required fields from version to meet the needs of the query. This would simplify the functions signature.